### PR TITLE
[CI Visibility] Refactor gotesting integration to be used by orchestrion

### PR DESCRIPTION
**Pull request by tonyredondo**
### What does this PR do?

This PR is a refactor of the existing code so the new methods can be used by the orchestrion instrumentation. It doesn't introduce any new functionality and the existing tests are enough for testing the refactored code.

### Motivation

Playing with orchestrion I realised that the current internal apis are really hard to call from an orchestrion integration, the refactor creates a new set of methods that can be easily called from methods prepend code kind of integrations. 

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [ ] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: internal/civisibility/integrations/gotesting/testingB.go [hunk 1]
```diff
@@ -33,6 +28,12 @@ var (
 
 	// subBenchmarkAutoNameRegex is a regex pattern to match the sub-benchmark auto name.
 	subBenchmarkAutoNameRegex = regexp.MustCompile(`(?si)\/\[DD:TestVisibility\].*`)
+
+	// civisibilityBenchmarksFuncs holds a map of *func(*testing.B) for tracking instrumented functions
+	civisibilityBenchmarksFuncs = map[*func(*testing.B)]bool{}
```
Comment (ManuelPalenzuelaDD) [inline internal/civisibility/integrations/gotesting/testingB.go:33]: nit: I think using a `map[*func(*testing.B)]struct{}` could be more efficient as a hashset datastructure?

**Review by ManuelPalenzuelaDD — APPROVED** [review]
LGTM, left a minor nit

File: internal/civisibility/integrations/gotesting/testingB.go (hunk 1 above)
Comment (tonyredondo) [reply internal/civisibility/integrations/gotesting/testingB.go]: done in [6a4ca88](https://github.com/DataDog/dd-trace-go/pull/2843/commits/6a4ca887a52919f7f3673b33f19aafb2356dbe65)

**Comment by eliottness** [comment]
Merging on behalf of @tonyredondo

**Pull request merged**